### PR TITLE
Simplify downstream integration

### PR DESCRIPTION
**Pull request by bhess**
- adds META.yml files
- adds new api: crypto_sign_signature and crypto_sign_verify
- adds namespacing
- converts CRLF to LF
- fixes for big-endian
- changes types from 'unsigned long long' to size_t
- fixes for strict compilers
- ensures newlines at end of files

### CODE REVIEW

File: src/mayo_1/api.c [hunk 1]
```diff
@@ -3,35 +3,55 @@
...
+#define MAYO_PARAMS 0
+#endif
+
 int
 crypto_sign_keypair(unsigned char *pk, unsigned char *sk) {
-    return mayo_keypair(&MAYO_1, pk, sk);
+    return mayo_keypair(MAYO_PARAMS, pk, sk);
 }
 
 #ifndef PQM4
```
Comment (mkannwischer) [inline src/mayo_1/api.c]: I think this is no longer needed if you change to `size_t` everywhere.

Comment (mkannwischer) [reply src/mayo_1/api.c]: Tested it. Yes, you can simply remove the #else part

**Review by mkannwischer — CHANGES_REQUESTED** [review]
Thanks @bhess for the integration into liboqs - that's going to be very useful for many people. The changes look mostly good to me. 

Tested this PR downstream in https://github.com/PQCMayo/MAYO-M4. 
It only requires some minor changes: https://github.com/PQCMayo/MAYO-M4/tree/liboqs-integration

Since you changed to `size_t`, you can remove the cases in the `api.c`

**Comment by bhess** [comment]
Thanks @mkannwischer, I removed the PQM4 cases from api.c/.h.

**Pull request merged**